src/analysis/stack_or_heap_enclosing.ml

```diff
@@ -10,20 +10,41 @@ type stack_or_heap =
 
 type stack_or_heap_enclosings = (Location.t * stack_or_heap) list
 
-let from_nodes ~pos ~path =
-  let aux (_env, node, _tail) =
+let from_nodes ~lsp_compat ~pos ~path =
+  let[@tail_mod_cons] rec tails = function
+    | hd :: tl -> (hd, tl) :: tails tl
+    | [] -> []
+  in
```

Comment:
This function seems a bit more general than necessary - I think it might be a bit clearer if it were something like:
```
let[@tail_mod_cons] rec with_parents = function
| hd :: next :: tl -> (hd, Some next) :: tails (next :: tl)
| [ hd ] -> [hd, None]
| [] -> []
```